docs/user/usage.md

```diff
@@ -73,12 +122,67 @@ may be required instead (see [custom validation for columns](#custom-column-vali
 
 ## Row steps
 
+Row steps are steps that can be declared for the Phase to run on the data one row at a time. Row steps can be 
+custom-written to do any logic that requires on the data in one row at a time.  This example finds the hour of day
+from a column that has already been declared as a phaser.DateTimeColumn.
+
+```python
+@row_step
+def find_hour_bucket(row, context):
+    row['hour'] = row['time'].hour
+    return row
+
+```
+
+Although row steps aren't meant to consider data from multiple rows, they can use data from other sources or from 
+the context. This simple example can be used to copy a required field that may be constant for the file as a whole into 
+every row for when that row is imported into a database of mixed sources.  
+
+```python
+@row_step
+def fill_user_name(row, context):
+    # When run on a file, this pipeline sets up a context with the user_name of the user who submitted the file   
+    row['contributor_user_name'] = context.get('user_name')
+    return row
+```
+
 ## Batch steps
 
 ## DataFrame steps
 
 ## Built-in steps
 
+### JSON utilities: flatten_dict and flatten_all
+
+These two built-in steps are helpful in transforming nested JSON data into values that can be directly processed.
+
+```python
+import phaser
+#Source data sample row:
+#    {'time': "2024-11-18 19:10:29+00:00",
+#     'value': 8.77018,
+#     'units': 'mmol/L',
+#     'payload': {
+#         'timestamp': 532858229,
+#         'logIndices': ['123456'],
+#         'rate': -1,
+#         'rssi': -90
+#     }
+#    }    
+
+class FlattenPhase(phaser.Phase):
+    steps = [
+        phaser.flatten_dict('payload')
+    ]
+
+# Phase output
+# time, value, units, payload__timestamp, payload__logIndices, payload__rate, payload__rssi
+# "2024-11-18 19:10:29+00:00", 8.77018, "mmol/L", 532858229, ['123456'], -1, -90
```

Comment:
This is an interesting example, because of that list value for `logIndices`.  If the list is `['123', '345', '567']`, does the output csv end up looking like this:

    "2024-11-18 19:10:29+00:00", 8.77018, "mmol/L", 532858229, ['123', '345', '567'], -1, -90

or this:

    "2024-11-18 19:10:29+00:00", 8.77018, "mmol/L", 532858229, "['123', '345', '567']", -1, -90

The second is certainly desired.  It may be desired to put double quotes around the list, no matter how many elements are in it.

Reply:
I added a test that confirms this is the behavior in test_csv.py